addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/discovery/YamahaDiscoveryParticipant.java

```diff
@@ -53,10 +53,15 @@ public DiscoveryResult createResult(RemoteDevice device) {
         } catch (Exception e) {
             // ignore and use the default label
         }
-        properties.put((String) YamahaReceiverBindingConstants.CONFIG_HOST_NAME,
-                device.getIdentity().getDescriptorURL().getHost());
-        properties.put((String) YamahaReceiverBindingConstants.CONFIG_ZONE,
-                YamahaReceiverCommunication.Zone.Main_Zone.name());
+
+        URL url = device.getIdentity().getDescriptorURL();
+        int port = url.getPort() == -1 ? 80 : url.getPort();
+        // Fix for upnp implementations on 8080
+        if (port == 8080) {
+            port = 80;
+        }
+
```

Comment:
File contains some empty lines that are not necessary. Please check formatting here and also in other classes.